include/zim/blob.h

```diff
@@ -35,7 +35,7 @@ namespace zim
    *
    * All `Blob`'s methods are threadsafe.
    */
-  class Blob
+  class DECLSPEC Blob
```

Comment:
`DECLSPEC` is not a good name for this macro. It doesn't convey its meaning well enough copying the underlying syntax from the least popular compiler used for building libzim. Besides macros in public headers must be named in such a way as to minimize probability of name clashes with other libraries. Below are the names of analogous macros from other projects:

- zstd: `ZSTDLIB_API`
- liblzma: `LZMA_API_EXPORT`
- zlib: `ZEXPORT`
- icu: `U_EXPORT`
- libcurl: `CURL_EXTERN`
- docopt: `DOCOPT_API`

By analogy, a good name for `libzim` should be `LIBZIM_API` (which is a better alternative to the other contender `LIBZIM_EXPORT` due to being shorter).

Reply:
@kelson42

Reply:
changed to LIBZIM_API